### PR TITLE
Refactor dhcp-spec unit-test

### PR DESCRIPTION
**Pull request by jlongever**
- Remove rabbitmq-server dependency.

@RackHD/corecommitters @zyoung51 @keedya @uppalk1 @tannoa2

### CODE REVIEW

File: spec/lib/protocol/dhcp-spec.js [hunk 1]
```diff
@@ -41,301 +53,242 @@ describe("DHCP protocol functions", function () {
...
 
     describe("removeLease", function () {
         it("should subscribe and receive removeLease results", function () {
-            var self = this,
-                testMac = "00:11:22:aa:bb:ee",
-                testResult = true;
-
-            return self.dhcp.subscribeRemoveLease(function (_data) {
+            var testResult = true;
+            messenger.subscribe = sinon.spy(function(a,b,callback,d) {
```
Comment (yyscamper) [inline spec/lib/protocol/dhcp-spec.js]: it's better not use such simple variable name, like `a`, `b`, `c`, ...

**Comment by yyscamper** [comment]
:+1:

**Comment by keedya** [comment]
+1

**Comment by jlongever** [comment]
test this please

**Pull request merged**